scripts/installer/windows/scripts/generate_dependency.py

```diff
@@ -0,0 +1,84 @@
+import ast
+import re
+import subprocess
+import copy
+from pathlib import Path
+from promptflow._sdk._utils import render_jinja_template
+
+
+def extract_requirements(file_path):
+    with open(file_path, 'r') as file:
+        tree = ast.parse(file.read())
+
+    install_requires = []
+    extras_requires = {}
+    for node in ast.walk(tree):
+        if isinstance(node, ast.Assign) and node.targets[0].id == 'REQUIRES':
+            install_requires = [elt.s for elt in node.value.elts]
+        elif isinstance(node, ast.Call) and getattr(node.func, 'id', None) == 'setup':
+            for keyword in node.keywords:
+                if keyword.arg == 'extras_require':
+                    extras_requires = ast.literal_eval(keyword.value)
+    return install_requires, extras_requires
+
+
+def extract_package_names(packages):
+    package_names = []
+    for package in packages:
+        match = re.match(r'^([a-zA-Z0-9-_.]+)', package)
+        if match:
+            package_names.append(match.group(1))
+    return package_names
+
+
+def get_package_dependencies(package_name_list):
+    dependencies = []
+    for package_name in package_name_list:
+        result = subprocess.run(['pip', 'show', package_name], stdout=subprocess.PIPE)
+        lines = result.stdout.decode('utf-8', errors="ignore").splitlines()
+        for line in lines:
+            if line.startswith('Requires'):
+                dependency = line.split(': ')[1].split(', ')
+                if dependency != ['']:
+                    dependencies.extend(dependency)
+                break
+    return dependencies
+
+
+if __name__ == '__main__':
+    dependencies = []
+    install_requires, extras_requires = extract_requirements('../../../../src/promptflow/setup.py')
```

Comment:
can this process still work when we done the layering sdk change? (separate pacakges)

Reply:
I think it depends on setup.py structure.

Reply:
we might plan to use pyproject.toml for new promptflow packages, instead of setup.py
here is an example for that: <https://github.com/microsoft/promptflow/pull/2268/files>
I understand it's not a big change to be compatible with that, might be aware of that.